src/visitable.cpp

```diff
@@ -830,8 +825,16 @@ long visitable<Character>::charges_of( const std::string &what, long limit ) con
 
     if( what == "UPS" ) {
         long qty = 0;
-        qty = sum_no_wrap( qty, charges_of( "UPS_off" ) );
-        qty = sum_no_wrap( qty, long( charges_of( "adv_UPS_off" ) / 0.6 ) );
+        if( p && p->has_item_with_flag( "UPS" ) ) {
+            for( auto &i : p->all_items_with_flag( "UPS" ) ) {
+                qty = sum_no_wrap( qty, charges_of( i->typeId() ) );
```

Comment:
> BevapDin on Sep 25  
This may count items several times: image a player with two UPS (of same type, say "UPS_off") in their inventory. The loop will go through both of them (all_items_with_flag returns two entries), so this line will be executed twice. And i->typeId() will be the same each time, so you are adding "all charges of items of type 'UPS_off'" twice.

 > Why are you not using the charges of the items you iterate over directly?